### DISCUSSION TITLE
Welcome to com.anupackages.debugconsole Discussions!

### DISCUSSION THREAD

**Discussion by ANU-CHEEKI-BREEKI** [ANU-CHEEKI-BREEKI/com.anupackages.debugconsole#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by HyagoOliveira** [comment]
Hi, my name is Hyago and i'm from Brazil.
Nice work in the project! The Console looks great for the time I've been using it. I can see you put a lot of care in this project. 👍🏽 

Now, I have some questions that I didn't find on the README How to Use section:
1. How the best way to instantiate de console?
2. We just have to put the DebugConsole prefab on Scene and create a button to call the Open function?
3. Is there even a simpler way to do it?

**Reply by ANU-CHEEKI-BREEKI** [reply]
Hi, glad to see you're enjoying the Console! And thanks for the questions! It will help me improve the README, and I'll do that as soon as possible.

I prefer to instantiate the Console at runtime from some bootstrap script. Usually, I'm using Zenject in my projects. This makes scene startup easier in general because I don't need to start the game from a specific scene, all manager scripts, and of course, the console are created from any scene from which I launch the game in the editor.

But _**just dragging the prefab to the scene is a completely valid way to start working with the console.**_

**_To open the console, use the `~` key._** Kind of like the default console key in any game I've played. I assumed this is the most intuitive and convenient way, so I even forgot to write this in the README :) Probably, I need to expose this key in the prefab settings.

`DebugConsole.Open()` or `DebugConsole.Close()` can be useful in some cases when you want to show or hide the console during the execution of some command. 
For example, screenshots for the console were rendered in Unity. I started the rendering process from a custom console command, which hides the console, takes screenshots of each predefined scene area one by one, and then enables the console again.

**Reply by HyagoOliveira** [reply]
Oh, I see.
I'm using the new Input System, so cannot open te console using the Tilde key.
I can do this feature. Are you open to contribution?

**Reply by ANU-CHEEKI-BREEKI** [reply]
You are right

I have plans to add a some switch to support only the new InputSystem. 
As a quick solution, you can set "Active input handling" to "Both" in the Player Settings. It should work

![image](https://github.com/ANU-CHEEKI-BREEKI/com.anupackages.debugconsole/assets/15821105/235d339c-560b-498e-8d87-53e048034991)

Unfortunately, I don't have experience in "open-source development" yet, so I can't guarantee smooth collaboration.
